gprofiler/profilers/python.py

```diff
@@ -151,6 +153,49 @@ def __init__(
         self.output_path = Path(self._storage_dir) / f"pyperf.{random_prefix()}.col"
         self.user_stacks_pages = user_stacks_pages
 
+    @staticmethod
+    def _fix_kernel_headers_symlink():
+        try:
+            if get_run_mode() not in ("container", "k8s"):
+                return
+
+            try:
+                uname = subprocess.check_output(["uname", "-r"]).decode('latin-1').strip()
+            except subprocess.CalledProcessError:
+                return
+
+            build_path = f"/lib/modules/{uname}/build"
+            if not os.path.islink(build_path):
+                return
+
+            if os.path.lexists(build_path + "/"):
+                return
+
+            lib_modules_uname_path = f"/lib/modules/{uname}"
+
+            os.makedirs('/dev/shm/modules_mount/upper', 555)
+            os.makedirs('/dev/shm/modules_mount/workdir', 555)
+            subprocess.check_call(
+                [
+                    'mount',
+                    '-t',
+                    'overlay',
+                    'overlay',
+                    '-o',
+                    f'upperdir=/dev/shm/modules_mount/upper,lowerdir={lib_modules_uname_path}'
+                    ',workdir=/dev/shm/modules_mount/workdir',
+                    lib_modules_uname_path,
+                ]
+            )
+
+            os.chmod(lib_modules_uname_path, 0o755)
+            link_target = os.readlink(build_path)
+            new_target = re.sub(r'(\.\./)+usr/src', '/usr/src', link_target, count=1)
```

Comment:
Don't we want to count the number of `../` and make sure they are what we expect (4, I suppose)? Or is there an hidden assumption here that the `/usr/src` part will always be the root?

Reply:
Assumption.

If the kernel headers actually exist in `/usr/src/`, we can point there just fine.
Otherwise, we probably won't have access from the container any way.

Reply:
Okay. Can comment the assumption

Reply:
done